pennylane/workflow/execution.py

```diff
@@ -200,18 +209,17 @@ def cost_fn(params, x):
         derivative_order=max_diff,
         executor_backend=executor_backend,
     )
-    config = _resolve_execution_config(config, device, tapes, transform_program=transform_program)
+    config = _resolve_execution_config(config, device, tapes)
 
     transform_program = transform_program or qml.transforms.core.TransformProgram()
-    transform_program, inner_transform = _setup_transform_program(
+    outer_transform, inner_transform = _setup_transform_program(
         transform_program, device, config, cache, cachesize
     )

```

Comment:
Since `_setup_transform_program` is also private, I'm tempted to remove `transform_program` from it too. So that then the call signature is just `_setup_transform_program(device, config, cache, cachesize)`.

Reply:
resolved since it's already done